presto-main/src/main/java/com/facebook/presto/operator/DeleteOperator.java

```diff
@@ -121,8 +121,9 @@ public void finish()
     {
         if (state == State.RUNNING) {
             state = State.FINISHING;
-            finishFuture = pageSource().finish();
-            requireNonNull(finishFuture, "finishFuture is null");
+            CompletableFuture<Collection<Slice>> future = pageSource().finish();
+            requireNonNull(future, "future is null");
```

Comment:
This isn't needed, as `toListenableFuture` will do the check. So it can be inlined
```java
finishFuture = toListenableFuture(pageSource().finish());